contracts/promisevalidator.sol

```diff
@@ -0,0 +1,86 @@
+pragma solidity ^0.4.0;
+
+import "./abstracts/witnessabstract.sol";
+import "./owned.sol";
+
+contract PromiseValidator is WitnessAbstract,Owned {
+
+  struct promise{
+    address beneficiary;
+    uint256 blockNumber;
+    uint8 sig_v;
+    bytes32 sig_r;
+    bytes32 sig_s;
+    bool exist;
+  }
+  uint GRACE_PERIOD = 35;//grace period of 35 blocks
+  //map serviceId to map clientAddress to promise
+  mapping(bytes32 => mapping(address=>promise))  promises;
+  //map caseid to pending time
+  mapping(bytes32 => uint)  gracePeriods;
+
+
+  function PromiseValidator() {
+	}
+
+  function expiered(bytes32 id) private returns (bool){
+    if (gracePeriods[id]!=0){
```

Comment:
I think grace periods should be specified as the rule of the game not the witness.
I should be in TransactitonAbstractContract and instnatiated in MirrorTransitions